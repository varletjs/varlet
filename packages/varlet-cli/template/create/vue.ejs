<template>
  <div class="var-<%- kebabCaseName %>"></div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import './<%- kebabCaseName %>.less'

export default defineComponent({
  name: '<%- namespace + bigCamelizeName %>',
})
</script>
