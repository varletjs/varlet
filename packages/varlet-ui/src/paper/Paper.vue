<template>
  <div
    :class="
      classes(
        n(),
        n('$--box'),
        n(`$-elevation--${elevation}`),
        [onClick, n('--cursor')],
        [round, n('--round')],
        [inline, n('$--inline-flex')]
      )
    "
    :style="{
      width: toSizeUnit(width),
      height: toSizeUnit(height),
      'border-radius': toSizeUnit(radius),
    }"
    v-ripple="{ disabled: !ripple }"
    @click="handleClick"
  >
    <slot />
  </div>
</template>

<script lang="ts">
import Ripple from '../ripple'
import { defineComponent } from 'vue'
import { call, createNamespace } from '../utils/components'
import { props } from './props'
import { toSizeUnit } from '../utils/elements'

const { n, classes } = createNamespace('paper')

export default defineComponent({
  name: 'VarPaper',
  directives: { Ripple },
  props,
  setup(props) {
    const handleClick = (e: Event) => {
      call(props.onClick, e)
    }

    return {
      n,
      classes,
      toSizeUnit,
      handleClick,
    }
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import '../styles/elevation';
@import './paper';
</style>
