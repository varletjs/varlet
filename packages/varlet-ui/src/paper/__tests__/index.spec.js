import Paper from '..'
import VarPaper from '../Paper'
import { mount } from '@vue/test-utils'
import { createApp } from 'vue'

test('test paper use', () => {
  const app = createApp({}).use(Paper)
  expect(app.component(Paper.name)).toBeTruthy()
})

test('test paper elevation', () => {
  expect(mount(VarPaper, { props: { elevation: 2 } }).html()).toMatchSnapshot()
})

test('test paper size', () => {
  expect(mount(VarPaper, { props: { width: 100, height: 100 } }).html()).toMatchSnapshot()
})

test('test paper round', () => {
  expect(mount(VarPaper, { props: { round: true } }).html()).toMatchSnapshot()
})

test('test paper inline', () => {
  expect(mount(VarPaper, { props: { inline: true } }).html()).toMatchSnapshot()
})

test('test paper radius', () => {
  expect(mount(VarPaper, { props: { radius: 10 } }).html()).toMatchSnapshot()
})

test('test paper onClick', () => {
  const onClick = jest.fn()

  const wrapper = mount(VarPaper, {
    props: {
      onClick,
    },
  })

  wrapper.trigger('click')
  expect(onClick).toBeCalledTimes(1)
})
