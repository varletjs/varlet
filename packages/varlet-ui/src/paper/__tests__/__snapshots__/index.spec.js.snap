// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test paper elevation 1`] = `"<div class=\\"var-paper var--box var-elevation--2\\"></div>"`;

exports[`test paper inline 1`] = `"<div class=\\"var-paper var--box var--inline-flex\\"></div>"`;

exports[`test paper radius 1`] = `"<div class=\\"var-paper var--box\\" style=\\"border-radius: 10px;\\"></div>"`;

exports[`test paper round 1`] = `"<div class=\\"var-paper var--box var-paper--round\\"></div>"`;

exports[`test paper size 1`] = `"<div class=\\"var-paper var--box\\" style=\\"width: 100px; height: 100px;\\"></div>"`;
